### PR TITLE
Secured RSA key generation against side-channel attacks

### PR DESCRIPTION
**Pull request by sam1013**
Replaced variable-time GCD with consttime inversion to avoid side-channel attacks on RSA key generation

### CODE REVIEW

**Comment by FdaSilvaYY** [comment]
Ready ? or this PR needs more than two approvals ;)

File: crypto/rsa/rsa_gen.c [hunk 1]
```diff
@@ -88,16 +89,26 @@ static int rsa_builtin_keygen(RSA *rsa, int bits, BIGNUM *e_value,
...
     for (;;) {
         if (!BN_generate_prime_ex(rsa->p, bitsp, 0, NULL, NULL, cb))
             goto err;
         if (!BN_sub(r2, rsa->p, BN_value_one()))
             goto err;
-        if (!BN_gcd(r1, r2, rsa->e, ctx))
-            goto err;
-        if (BN_is_one(r1))
+        ERR_set_mark();
+        if (BN_mod_inverse(r1, r2, rsa->e, ctx) != NULL) {
```
Comment (kroeckx) [inline crypto/rsa/rsa_gen.c]: #5161 calls BN_set_flags(r2, BN_FLG_CONSTTIME).

Comment (mattcaswell) [reply crypto/rsa/rsa_gen.c]: So does this - before the "for"

**Comment by mattcaswell** [comment]
@kroeckx - are you ok for us to push this one too (I just pushed the master version)?

**Comment by kroeckx** [comment]
@mattcaswell: Will you merge this?

**Comment by mattcaswell** [comment]
Pushed to 1.1.0 and 1.0.2.

**Pull request closed without merging**